PGD: clarify bdr.join_node_group()'s synchronize_structure parameter

ibarwick opened the pull request:
## What Changed?

Current state: https://www.enterprisedb.com/docs/pgd/latest/reference/nodes-management-interfaces/#bdrjoin_node_group

This patch:
- explicitly states the default value (`all`) -> see https://github.com/EnterpriseDB/bdr/blob/5a247059b06984a7e2d50dcaee7aaa58abc2720f/bdr_functions.c#L2358
- notes that data is not synchronized for witness nodes -> see https://www.enterprisedb.com/docs/pgd/latest/node_management/witness_nodes/

BDR-4520.

djw-m commented on product_docs/docs/pgd/5/reference/nodes-management-interfaces.mdx:
```diff
@@ -245,9 +245,9 @@ bdr.join_node_group (
+    to do during the join. Valid options are `all` (the default), which synchronizes
     the complete database structure, and `none`, which doesn't synchronize any
-    structure. However, it still synchronizes data.
+    structure. However, it still synchronizes data (except for witness nodes).
```
```suggestion
    structure. However, it still synchronizes data (except for witness nodes, which by design do not synchronize data).
```

djw-m commented on product_docs/docs/pgd/5/reference/nodes-management-interfaces.mdx:
```diff
@@ -245,9 +245,9 @@ bdr.join_node_group (
     returning. Defaults to `true`.
 -   `synchronize_structure` &mdash; Set the kind of structure (schema) synchronization
-    to do during the join. Valid options are `all`, which synchronizes
+    to do during the join. Valid options are `all` (the default), which synchronizes
```
```suggestion
    to do during the join. The default setting is `all` (the default), which synchronizes
```

djw-m commented on product_docs/docs/pgd/5/reference/nodes-management-interfaces.mdx:
```diff
@@ -245,9 +245,9 @@ bdr.join_node_group (
 -   `synchronize_structure` &mdash; Set the kind of structure (schema) synchronization
-    to do during the join. Valid options are `all`, which synchronizes
+    to do during the join. Valid options are `all` (the default), which synchronizes
     the complete database structure, and `none`, which doesn't synchronize any
```
```suggestion
    the complete database structure. The other available setting is `none` which doesn't synchronize any
```

djw-m reviewed (commented):
Broke up the options to be more precise and added a touch of explanation to the "witness nodes" exception.

ibarwick commented:
> Broke up the options to be more precise and added a touch of explanation to the "witness nodes" exception.

Make sense. 

I couldn't see any changes to my original commit in the branch for this PR. I've added the suggested changes, rebased the branch onto `dev/pgd5.4` and force-pushed it.

LenzGr commented:
Thank you. Please resolve PR comments that you have addressed.

LenzGr reviewed (approved):
LGTM! Thank you.
